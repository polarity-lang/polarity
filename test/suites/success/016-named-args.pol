data Bool { True, False }

def Bool.and(other: Bool): Bool {
    True => other,
    False => False
}

data List {
    Nil,
    Cons(head: Bool, tail: List)
}

data Option(elem: Type) {
    None(elem: Type): Option(elem),
    Some(elem: Type, x: elem): Option(elem),
}

let example1 : List {
    Cons(head := True, tail := Nil)
}
let example2: Bool { True.and(other := False) }
let example3: Option(elem := Bool) { None(Bool) }
