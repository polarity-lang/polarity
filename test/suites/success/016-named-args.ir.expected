def .and(other) {
    True => other,
    False => False,
}

let example1 { Cons(True, Nil) }

let example2 { True.and(False) }

let example3 { None }
